### PR TITLE
Include demo pages in dev workflow.

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

This helps keep the demo pages in sync in the dev workflow. It has two main parts:

1. `demoPageSync.js` - watches changes to index and individual demo pages and will
    * copy over modified pages 
    * trigger a build
2. `demoPageReloader.js` - establishes a web socket connection to reload when a build finishes

@pr3tori4n @driskull @asangma 🥇bonus points if you can take this for a test drive and give any feedback.

### CODE REVIEW

**Comment by asangma** [comment]
Demo pages appear to be showing changes on save. Would be cool if changes in SASS partials triggered a rebuild.

Currently, when I change e.g. `src/assets/styles/_layout.scss` there is no rebuild. And even if I save another SASS file, it doesn't pick up changes in `_layout.scss`.

**Comment by jcfranco** [comment]
@asangma Thanks for testing. I'll submit a follow-up PR to tackle the sassness.

File: support/demoPageSync.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const projectRoot = normalize(`${__dirname}/../`);
+const sourceDir = `${projectRoot}src/`;
+const sourceDemoDir = `${sourceDir}demos/`;
+const destinationDir = `${projectRoot}www/`;
+const destinationDemoDir = `${destinationDir}demos/`;
+const sourceIndexFile = `${sourceDir}index.html`;
+const targetTouchFile = `${sourceDir}components/interfaces.ts`;
+const noop = () => {};
+
+watch(sourceDemoDir, (type, changedFile) => triggerBuild(sourceDemoDir, destinationDemoDir, changedFile));
```
[the remark this answers is not in the document]
Comment (pr3tori4n) [reply support/demoPageSync.js]: Might want to consider using [Chodikar](https://www.npmjs.com/package/chokidar#why). Fs.watch is known to have some quirks.

**Comment by jcfranco** [comment]
Thanks for testing. Will fix. #showwindowssomelove ❤️

**Pull request merged**